ios/Plugin/Plugin.swift

```diff
@@ -59,18 +58,24 @@ public class CapacitorCalendar: CAPPlugin {
                 event.location = location
                 event.notes = notes
                 event.calendar = calendar
-
                 event.startDate = eventStartDate
-                let duration = Int(endDate - startDate);
-                let moduloDay = duration % (60 * 60 * 24);
 
-                if (moduloDay == 0) {
-                    event.isAllDay = true;
-                    event.endDate = Date(timeIntervalSince1970: (endDate / 1000) - 1)
-                } else {
+                print ("Duration: \(duration), moduloDay \(moduloDay)")
+                if let allDay = call.getBool("allDay") {
```

Comment:
Indeed that's the goal. As @Fir3st asked :
> you can make your changes optional - if the parameter is provided, allDay is handled by that parameter, otherwise, it's handled the old way.

If `allDay` is `true/false` it enters the first block. If it's not defined it enters the else block.

Reply:
All right, thanks.